mlflow/utils/proto_json_utils.py

```diff
@@ -136,14 +136,45 @@ def _get_jsonable_obj(data, pandas_orient="records"):
         return data
 
 
-def parse_tf_serving_input(inp_dict):
+def parse_tf_serving_input(inp_dict, schema=None):
     """
     :param inp_dict: A dict deserialized from a JSON string formatted as described in TF's
                      serving API doc
                      (https://www.tensorflow.org/tfx/serving/api_rest#request_format_2)
+    :param schema: Mlflow schema used when parsing the data.
     """
     import numpy as np
 
+    def cast_schema_type(input_data):
+        if schema is not None:
+            if schema.has_input_names():
+                input_names = schema.input_names()
+                if len(input_names) == 1 and isinstance(input_data, list):
+                    # for schemas with a single column, match input with column
+                    input_data = {input_names[0]: input_data}
+                if not isinstance(input_data, dict):
+                    raise MlflowException(
+                        "This model contains a tensor-based model signature with input names, which"
```

Comment:
nit: I think the message should contain something like: Failed to parse, the input ins incompatible with the model signature, since in this context, we are parsing the input data rather than scoring the model .